airflow/contrib/hooks/hortonworks_ambari_hook.py

```diff
@@ -0,0 +1,131 @@
+                    which will be used to Hortonworks rest service
+        :type ambari_conn_id: str
+        """
+        super(HdpAmbariHook, self).__init__(ambari_conn_id)
+
+        self.headers = {
+            'Content-Type': "application/x-www-form-urlencoded"
+        }
+        connection = self.get_connection(ambari_conn_id)
+        self.cluster_name = str(connection.host).split("//")[0].split(".")[0]  # clustername
+        username = connection.login
+        self.params = {"user.name": username}
+
+    def submit_hive_job(self, datas: dict):
+        """
+        Executes hql code or hive script in Azure HDInsight Cluster
+
+        :param datas: parametres of Hive script
+        :type datas: object
+
+        """
+
+        self.method = "POST"
+        hive_endpoint = "templeton/v1/"
+        statement_non_terminated_status_list = ['INIT', 'RUNNING', 'PREP']
+        submit_endpoint = hive_endpoint + "hive"
+
+        self.log.debug("Submiting hive  Script: %s", str(datas))
+        response = self.run(endpoint=submit_endpoint, data=datas, headers=self.headers)
+
+        job_id = response["id"]
+        status_endpoint = hive_endpoint + "jobs/" + str(job_id)
+        self.log.info("Finished submitting hive script job_id: %s", job_id)
+
+        statements_state = 'INIT'
+        while statements_state in statement_non_terminated_status_list:
+
+            # todo: test execution_timeout
+            time.sleep(5)
+            try:
+                result = self._get_job_status(endpoint=status_endpoint)["status"]["state"]
+            except Exception:
+                pass
+
+            statements_state = result["status"]["state"]
+            statements_exit_value = result["exitValue"]
+            if statements_state is not None:
+                # Check submitted job's result
+                if statements_state in ['KILLED', 'FAILED'] or statements_exit_value != "0":
+                    result = "job failed. state: %s", statements_state
+                    self.log.error(result)
+                    raise AirflowException(result)
+                else:
+                    self.log.debug("Checking Hive job: %s", statements_state)
+
+        self.log.debug("Statement %s ", statements_state)
+        self.log.debug("Finished executing WebHCatHiveSubmitOperator")
+
+    def submit_spark_job(self, datas):
+        """
+        Submit spark job or spark-sql to Azure HDInsight Cluster
+
+        :param datas: parametres of Spark Job
+        :type datas: object
+        """
+
+        statement_non_terminated_status_list = ['INIT', 'starting', 'running', "waiting", "available"]
+        spark_endpoint = "livy/batches"
+        submit_endpoint = spark_endpoint
+        self.method = "POST"
+        datas["user.name"] = self.username
+
+        self.log.info("Submiting spark Job: %s ", json.dumps(datas))
+        response = self.run(endpoint=submit_endpoint, data=json.dumps(datas))
```

Comment:
Aren't headers missing here? They might be necessary in order to satisfy Ambary CSRF checks